modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ParameterDslDeclarationEnricher.java

```diff
@@ -65,7 +75,32 @@ protected void onParameter(ParameterGroupDeclaration parameterGroup, ParameterDe
           ParameterDslConfiguration.Builder builder = ParameterDslConfiguration.builder();
           boolean isContent = !declaration.getRole().equals(ParameterRole.BEHAVIOUR);
           ParameterDslConfiguration dslConfiguration = declaration.getDslConfiguration();
-          declaration.getType().accept(new MetadataTypeVisitor() {
+
+          MetadataType normalizedDeclarationType = extensionDeclaration.getTypes().stream()
+              .filter(type -> getId(type).equals(getId(declaration.getType())))
+              .findFirst()
+              .map((MetadataType type) -> {
+                if (type instanceof ObjectType) {
+                  final Map<Class<? extends TypeAnnotation>, TypeAnnotation> normalizedAnnotationsByClass =
+                      type.getAnnotations().stream()
+                          .collect(toMap(ann -> ann.getClass(), Functions.identity()));
+
+                  declaration.getType().getAnnotation(ClassInformationAnnotation.class)
+                      .ifPresent(paramClassInfo -> normalizedAnnotationsByClass.put(ClassInformationAnnotation.class,
+                                                                                    paramClassInfo));
+                  declaration.getType().getAnnotation(ParameterDslAnnotation.class)
+                      .ifPresent(paramClassInfo -> normalizedAnnotationsByClass.put(ParameterDslAnnotation.class,
+                                                                                    paramClassInfo));
+                  return new DefaultObjectType(((ObjectType) type).getFields(), ((ObjectType) type).isOrdered(),
```

Comment:
what happens if the type is a dynamic one and the ClassInformation annotation simply points to `java.lang.Object`